templates/default/kafka-consul.hcl.erb

```diff
@@ -1,16 +1,34 @@
-service {
-    id = "kafka/monitoring"
-    name = "kafka"
-    tags = [
-        "monitoring"
-    ]
-    port = <%= node['kkafka']['metrics_port'] %> 
-    check = {
-        id = "kafka-check"
-        name = "Health check for Kafka"
-        http = "http://localhost:<%= node['kkafka']['metrics_port'] %>/metrics"
-        interval = "20s"
-        timeout = "20s"
-        status = "passing"
-    }
-} 
\ No newline at end of file
+services = [
+    {
+        id = "kafka/monitoring"
+        name = "kafka"
+        tags = [
+            "broker"
+        ]
+        port = <%= node['kkafka']['broker']['port'] %>
+        check = {
+            id = "kafka-check"
+            name = "Health check for Kafka"
+            http = "http://localhost:<%= node['kkafka']['metrics_port'] %>/metrics"
+            interval = "20s"
+            timeout = "20s"
+            status = "passing"
+        }
+    },
+    {
+        id = "kafka/monitoring"
+        name = "kafka"
+        tags = [
+            "monitoring"
+        ]
+        port = <%= node['kkafka']['metrics_port'] %>
+        check = {
```

Comment:
Since it's the same check with the broker, you can reference the health check as we do in https://github.com/logicalclocks/hops-hadoop-chef/blob/master/templates/default/consul/rm-consul.hcl.erb#L30